SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -3531,6 +3540,64 @@ bool USpatialNetDriver::HasTimedOut(const float Interval, uint64& TimeStamp)
 	return false;
 }
 
+void USpatialNetDriver::ServerReplicateActors_BuildConsiderList(TArray<FNetworkObjectInfo*>& OutConsiderList, const float ServerTickTime)
+{
+	const bool bSpatialNetworking = GetDefault<UGeneralProjectSettings>()->UsesSpatialNetworking();
+
+	if (!bSpatialNetworking || AuthorityDebugger == nullptr)
+	{
+		// Default case to only consider auth actors for replication
+		Super::ServerReplicateActors_BuildConsiderList(OutConsiderList, ServerTickTime);
+		return;
+	}
+
+	// Spatial authority debugger case to check non-auth actors for invalid changes and then add auth actors to the consider list for
+	// replication
+	TArray<FNetworkObjectInfo*> TmpConsiderList;
+	TmpConsiderList.Reserve(GetNetworkObjectList().GetActiveObjects().Num());
+	Super::ServerReplicateActors_BuildConsiderList(TmpConsiderList, ServerTickTime);
+
+	for (FNetworkObjectInfo* const& ActorInfo : TmpConsiderList)
+	{
+		AActor* Actor = ActorInfo->Actor;
+
+		if (Actor->HasAuthority())
+		{
+			//  Only add auth actors only to the consider list
+			OutConsiderList.Add(ActorInfo);
+		}
+		else
+		{
+			// Check for unauthorised data changes to non-auth actors
+			AuthorityDebugger->CheckUnauthorisedDataChanges(*Actor);
+		}
+	}
+}
+
+void USpatialNetDriver::AddSpatialShadowActor(const Worker_EntityId_Key EntityId)
+{
+	if (AuthorityDebugger != nullptr)
```

Comment:
I made this comment elsewhere, so just briefly, do we think its worthwhile excluding this code from shipping/testing builds? I see no value in running this code in either builds.